packages/go_router_builder/example/lib/main.g.dart

```diff
@@ -43,9 +43,11 @@ extension $HomeRouteExtension on HomeRoute {
         '/',
       );
 
-  void go(BuildContext context) => context.go(location, extra: this);
+  void go(BuildContext context, {Object? extra}) =>
+      context.go(location, extra: extra);
 
-  void push(BuildContext context) => context.push(location, extra: this);
+  void push(BuildContext context, {Object? extra}) =>
```

Comment:
also they can use $extra to pass the value, we should probably not expose the extra here?

Reply:
Because passing value to $extra will not change the `GoRouterState.extra`  and this example is using it. 

https://github.com/flutter/packages/blob/0b681aa62a2ba40891f505c72f74f5856ce90300/packages/go_router_builder/example/lib/main.g.dart#L86

Reply:
Done